### PR TITLE
Dhp 937 fix broken tests

### PR DESCRIPTION
**Pull request by jpresley23-tw**
## Summary

- We renamed the dhp tests so they run on GitHub Actions
- We fixed broken tests

## Testing done

-  We ran the unit tests for dhp and saw they passed
- We tested the unit tests to ensure they fail when they should fail

### Quality Assurance & Testing

- [ x ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).

### CODE REVIEW

File: src/applications/dhp-connected-devices/tests/components/TermsAndConditions.unit.spec.jsx [hunk 1]
```diff
@@ -19,14 +19,17 @@ describe('Terms and Conditions', () => {
     expect(screen.getByTestId('fitbit-terms-and-conditions')).to.exist;
     expect(screen.getByTestId('fitbit-terms-and-conditions-content')).to.exist;
     expect(
-      screen.getByText('Sign into Fitbit and share your Fitbit data with VA'),
-    ).to.exist;
-    expect(screen.getByText('View Terms')).to.exist;
+      screen.getByTestId('fitbit-terms-and-conditions').textContent,
+    ).to.have.string('Sign into Fitbit and share your Fit');
```
Comment (yonidoronpeters) [inline src/applications/dhp-connected-devices/tests/components/TermsAndConditions.unit.spec.jsx:23]: Should we use the whole text here? 
```suggestion
    ).to.have.string('Sign into Fitbit and share your Fitbit data with VA');
```

**Review by yonidoronpeters — APPROVED** [review]
Looks great, thanks for fixing the tests and make them run in CI!

File: src/applications/dhp-connected-devices/tests/components/DeviceDisconnectionCard.unit.spec.jsx [hunk 2]
```diff
@@ -41,9 +42,11 @@ describe('Device disconnection card', () => {
...
     it('Renders vendor terms and conditions', () => {
-      expect(screen.getByTestId('test-vendor-terms-and-conditions')).to.exist;
-      expect(screen.getByTestId('test-vendor-terms-and-conditions-content')).to
-        .exist;
+      const card = renderInReduxProvider(
+        <DeviceDisconnectionCard device={{ name: 'Fitbit', key: 'fitbit' }} />,
+      );
+      expect(card.getByTestId('fitbit-terms-and-conditions')).to.exist;
+      expect(card.getByTestId('fitbit-terms-and-conditions-content')).to.exist;
     });
```
Comment (jpresley23-tw) [inline src/applications/dhp-connected-devices/tests/components/DeviceDisconnectionCard.unit.spec.jsx:50]: To test this fails when it's supposed to, remove `<TermsAndCondidtions` component in the `DeviceDisconnectionCard` component.

File: src/applications/dhp-connected-devices/tests/components/TermsAndConditions.unit.spec.jsx [hunk 3]
```diff
@@ -19,14 +19,17 @@ describe('Terms and Conditions', () => {
...
     expect(
-      screen.getByText('Sign into Fitbit and share your Fitbit data with VA'),
-    ).to.exist;
-    expect(screen.getByText('View Terms')).to.exist;
+      screen.getByTestId('fitbit-terms-and-conditions').textContent,
+    ).to.have.string('Sign into Fitbit and share your Fitbit data with VA');
+    expect(
+      screen.getByTestId('fitbit-terms-and-conditions').innerHTML,
+    ).to.have.string('View Terms');
   });
```
Comment (jpresley23-tw) [inline src/applications/dhp-connected-devices/tests/components/TermsAndConditions.unit.spec.jsx:27]: To confirm these tests fail when expected, remove the strings from the `TermsAndConditions` component. Note the `View Terms` is a value for an attribute in a button element.

File: src/applications/dhp-connected-devices/tests/components/TermsAndConditions.unit.spec.jsx [hunk 4]
```diff
@@ -19,14 +19,17 @@ describe('Terms and Conditions', () => {
...
   it('Only renders Ts and Cs for device in TermsAndConditionsContentMap', () => {
     const screen = render(<TermsAndConditions device={deviceNoTsAndCs} />);
-    expect(screen.getByTestId('test-vendor-terms-and-conditions')).to.not.exist;
-    expect(screen.getByTestId('test-vendor-terms-and-conditions-content')).to
+    expect(screen.queryByTestId('test-vendor-terms-and-conditions')).to.not
+      .exist;
+    expect(screen.queryByTestId('test-vendor-terms-and-conditions-content')).to
       .not.exist;
   });
```
Comment (jpresley23-tw) [inline src/applications/dhp-connected-devices/tests/components/TermsAndConditions.unit.spec.jsx:34]: To confirm this fails when expected, remove the `if (device.key in TermsAndConditionsContentMap) {` clause in the `TermsAndConditions` component.

File: src/applications/dhp-connected-devices/tests/components/DeviceDisconnectionCard.unit.spec.jsx (hunk 2 above)
Comment (mariahsager-tw) [reply src/applications/dhp-connected-devices/tests/components/DeviceDisconnectionCard.unit.spec.jsx:50]: QA: Task failed successfully

File: src/applications/dhp-connected-devices/tests/components/TermsAndConditions.unit.spec.jsx (hunk 3 above)
Comment (mariahsager-tw) [reply src/applications/dhp-connected-devices/tests/components/TermsAndConditions.unit.spec.jsx:27]: QA: Task failed successfully

**Pull request merged**